qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,385 @@
+            for j in range(0, i):
+                qc.cx(qr[i], qr[j])
+            qc.x(qr[i])
+            qc.cx(qr[i], qr[i + 1])
+
+        # pylint: disable=unused-argument
+        def control(num_ctrl_qubits=1, label=None, ctrl_state=None):
+            qr_state = QuantumRegister(self.num_state_qubits + 1)
+            if self.num_state_qubits > 1:
+                qr_ancilla = AncillaRegister(max(1, self.num_state_qubits - 1))
+                qc_control = QuantumCircuit(qr_state, qr_ancilla, name='off_diags')
+            else:
+                qc_control = QuantumCircuit(qr_state, name='off_diags')
+                qr_ancilla = None
+            # Control will be qr[0]
+            q_control = qr_state[0]
+            qr = qr_state[1:]
+            # Gates for H2 with t
+            qc_control.cu(-2 * theta, 3 * np.pi / 2, np.pi / 2, 0, q_control, qr[0])
+
+            # Gates for H3
+            for i in range(0, self.num_state_qubits - 1):
+                q_controls = []
+                q_controls.append(q_control)
+                qc_control.cx(qr[i], qr[i + 1])
+                q_controls.append(qr[i + 1])
+
+                # Now we want controlled by 0
+                qc_control.x(qr[i])
+                for j in range(i, 0, -1):
+                    qc_control.cx(qr[i], qr[j - 1])
+                    q_controls.append(qr[j - 1])
+                qc_control.x(qr[i])
+
+                # Multicontrolled x rotation
+                if len(q_controls) > 1:
+                    ugate = UGate(-2 * theta, 3 * np.pi / 2, np.pi / 2)
+                    qc_control.append(MCMTVChain(ugate, len(q_controls), 1), q_controls[:] +
+                                      [qr[i]] + qr_ancilla[:len(q_controls)-1])
+                else:
+                    qc_control.cu(-2 * theta, 3 * np.pi / 2, np.pi / 2, 0, q_controls[0], qr[i])
+
+                # Uncompute
+                qc_control.x(qr[i])
+                for j in range(0, i):
+                    qc_control.cx(qr[i], qr[j])
+                qc_control.x(qr[i])
+                qc_control.cx(qr[i], qr[i + 1])
+            return qc_control
+
+        qc.control = control
+        return qc
+
+    def inverse(self):
+        return Tridiagonal(self.num_state_qubits, self.main_entry, self.off_diag,
+                           evo_time=-1 * self.evo_time)
+
+    def power(self, power: int, matrix_power: bool = False) -> QuantumCircuit:
+        """Build powers of the circuit.
+
+        Args:
+            power: The power to raise this circuit to.
+            matrix_power: If True, the circuit is converted to a matrix and then the
+                matrix power is computed. If False, and ``power`` is a positive integer,
+                the implementation defaults to ``repeat``.
+
+        Returns:
+            The quantum circuit implementing powers of the unitary.
+        """
+        qc_raw = QuantumCircuit(self.num_state_qubits)
+
+        # pylint: disable=unused-argument
```

Comment:
Is this because matrix_power is unused. Is there some intended future use - I see it defined by the base class - but I never see any use of it here

Reply:
This is some dummy circuit because only the `control()` method within `power()` is used.